src/components/jobManager.ts

```diff
@@ -327,4 +333,45 @@ export class JobManager {
 			uris,
 		});
 	}
+
+	public async refreshAllJobs(): Promise<void> {
+		const allJobs = Array.from(this.getJobs());
+		const allCodemods = await this.__engineService.getCodemodList();
+		const codemodHashUriTuple: [string, Uri][] = [];
+
+		allJobs.forEach((job) => {
+			if (job.oldUri === null) {
+				return;
+			}
+
+			const jobCodemod =  allCodemods.find(c => c.name === job.codemodName) ?? null;
+
+			if(jobCodemod === null) {
+				return;
+			} 
+			codemodHashUriTuple.push([jobCodemod.hashDigest, job.oldUri]);
+		});
+
+		const executionId = buildExecutionId();
+		const happenedAt = String(Date.now());
+
+		const messages: Message[] = codemodHashUriTuple.map(([codemodHash, uri]) =>({
```

Comment:
how does that work with the principle that only one execution can be managed?

Reply:
Currently we can have only 1 execution at a time, and we can only pass 1 codemod to the engine, right? 
I will need to add ability to `await` for execution to complete and then chain executions. 
Something like 
```
for(const codemod of codemods) {
await this.__engineService.executeCodemod(codemod, uris);
}
```